src/android.ts

```diff
@@ -290,6 +290,7 @@ export const androidApi = <RunTarget extends SupportedRunTarget<'android'>>(
         const { stdout } = await adb(['emu', 'avd', 'snapshot', 'load', snapshotName]);
         if (stdout.includes('KO')) throw new Error(`Failed to load snapshot: ${stdout}.`);
 
+        await this.waitForDevice();
         await this.ensureDevice();
```

Comment:
Wouldn’t it be more idiomatic now, to also remove `ensureDevice` from here and require the user to remeber this themselves? I am not sure.

Reply:
I don't think that would be more idiomatic. We removed the waiting from `ensureDevice()` because there are cases where you don't want that to wait. I can't think of a case where you would want to reset your device but then not have it be ready afterwards.